syntax = "proto3";

package user;

service UserService {
  rpc GetUserByToken (GetUserByTokenRequest) returns (UserResponse);
  rpc GetUserById (GetUserByIdRequest) returns (UserResponse);
}

message GetUserByTokenRequest {
  string token = 1;
}

message GetUserByIdRequest {
  int64 user_id = 1;
}

message UserResponse {
  bool is_valid = 1;
  string error_message = 2;
  int64 id = 3;
  string username = 4;
  string email = 5;
  int32 role = 6;
}
